GreengrassIPC.md

```diff
@@ -0,0 +1,172 @@
+# Greengrass IPC in the v2 Javascript SDK
+
+# Table of contents
+* [Developer Preview Disclaimer](#developer-preview-disclaimer)
+* [Cross-SDK API Differences](#cross-sdk-differences)
+* [Example](#greengrass-ipc-example)
+* [Greengrass IPC Client](#greengrass-ipc-client)
+  * [Client Setup](#client-setup)
+  * [Client Shutdown](#client-shutdown)
+  * [Client Events](#client-events)
+  * [Client Error Handling](#client-error-handling)
+* [Greengrass IPC Operations](#greengrass-ipc-operations)
+  * [Request-Response Operations](#request-response-operations)
+  * [Streaming Operations](#streaming-operations)
+  * [Operation Error Handling](#operation-error-handling)
+* [Miscellaneous Topics](#miscellaneous-topics)
+  * [Binary Data](#binary-data)
+
+Complete API documentation for Javascript Greengrass IPC can be found on [Github](https://aws.github.io/aws-iot-device-sdk-js-v2/node/modules/greengrasscoreipc)
+
+Greengrass IPC is intended for use in developing Greengrass components.  This guide does 
+not give any guidance on component development.  See the
+[Greengrass component](https://docs.aws.amazon.com/greengrass/v2/developerguide/greengrass-components.html) AWS 
+documentation for more details on that subject.
+
+This guide is a work in progress.  Feedback is welcomed.
+
+## Developer Preview Disclaimer
+Javascript Greengrass IPC support is currently in **developer preview**.  We encourage feedback at all times, but feedback during the preview window
+is especially valuable in shaping the final product.  During the preview period we may make backwards-incompatible changes to the
+public API.
+
+## Cross-SDK Differences
+The Javascript Greengrass IPC implementation has several differences relative to the IPC implementations in the other v2 SDKs:
+* No handler types - Javascript Greengrass IPC uses Javascript's event system (events, listeners, emitters) for all asynchronous notifications.
+* Single client - Other SDKs have both a v1 and v2 client, where the v1 client is low-level and more verbose (and complex) in usage, while the v2 client is streamlined and less prone to accidental misuse.  Javascript Greengrass IPC skips the v1 API and starts directly with the v2 client approach.
```

Comment:
Two extremely small nitpicks:

* `Other SDKs have both...` -> `Other v2 IoT SDKs have both...`
* `Javascript Greengrass IPC skips the v1 API and starts...` -> `Javascript Greengrass IPC skips the v1 client approach and starts...`